cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/preview/PreviewRunnerTwillRunnable.java

```diff
@@ -238,36 +243,46 @@ protected void configure() {
     }
 
     modules.add(new PreviewRunnerManagerModule().getDistributedModules());
-    modules.add(new DataSetServiceModules().getStandaloneModules());
-    modules.add(new DataSetsModules().getStandaloneModules());
-    modules.add(new AppFabricServiceRuntimeModule(cConf).getStandaloneModules());
-    modules.add(new ProgramRunnerRuntimeModule().getStandaloneModules());
-    modules.add(new MetricsStoreModule());
     modules.add(new MessagingClientModule());
-    modules.add(new AuditModule());
+    // We should consider if preview pods should be allowed to retrieve secrets from appfabric.
     modules.add(new SecureStoreClientModule());
-    modules.add(new MetadataReaderWriterModules().getStandaloneModules());
+    // Needed for InMemoryProgramRunnerModule. We use local metadata reader/publisher to avoid conflicting with
+    // metadata stored in AppFabric.
     modules.add(new DFSLocationModule());
-    modules.add(new MetadataServiceModule());
-    modules.add(new CoreSecurityRuntimeModule().getInMemoryModules());
+    // Configurator tasks should be executed in-memory since it is in the preview runner pod.
+    modules.add(new FactoryModuleBuilder().implement(Configurator.class, InMemoryConfigurator.class)
+                  .build(ConfiguratorFactory.class));
+
     modules.add(new AuthenticationContextModules().getMasterWorkerModule());
-    modules.add(new AuthorizationModule());
     modules.add(new AuthorizationEnforcementModule().getNoOpModules());
-    modules.add(Modules.override(
-      new DataFabricModules("master").getDistributedModules()).with(new AbstractModule() {
+    modules.add(new AbstractModule() {
       @Override
       protected void configure() {
-        // Bind transaction system to a constant one, basically no transaction, with every write become
-        // visible immediately.
-        // TODO: Ideally we shouldn't need this at all. However, it is needed now to satisfy dependencies
-        bind(TransactionSystemClientService.class).to(DelegatingTransactionSystemClientService.class);
         bind(TransactionSystemClient.class).to(ConstantTransactionSystemClient.class);
 
-        bind(ExploreClient.class).to(UnsupportedExploreClient.class);
         bind(PreviewRequestPollerInfoProvider.class).toInstance(() -> pollerInfoBytes);
-      }
-    }));
 
+        // Artifact Repository should use RemoteArtifactRepository.
+        // TODO: Consider adding a remote artifact respository handler to the preview manager so that preview runners
```

Comment:
Add JIRA ticket to the TODO so we don't forget about this.

Reply:
BTW, I think we added artifact localizer to preview so we can cache artifacts.

Reply:
Opened CDAP-19041 to track this. Indeed, artifact localizer is supported in preview already.